docs/Developer Documentation/Logger/index.md

```diff
@@ -0,0 +1,159 @@
+# SDL Logger
+
+Open source version of SDL is using log4cxx for logging.
+Because of lack of log4cxx stability OEM should have an ability to replace log4cxx logger with any other logging boost or syslog.
+
+
+SDL implements Logger abstration for easily replacement of logging library.
+
+|||
+High Level Design
+![Logger](./assets/high_level_design.png)
+|||
+
+
+All components have aceess to `Logger` interface and use it for logging.
+
+### Logger interface 
+
+Logger macroses use Logger interface for sending messages to External Logger. 
+Logger interface contains only methods required by any SDL component to perform logging : 
+
+ * instance() - singleton access
+ * PushLog(LogMessage)
+ * IsEnabledFor(LogLevel)
+ * DeInit()
+ * Flush() 
+
+
+## LoggerImpl
+
+`Logger` interface is implemented by `LoggerImpl`. 
+
+`LoggerImpl` use message loop thread to proxy log messages to third party (external) logger.
+`LoggerImpl` owns `ThirdPartyLoggerInterface` and controls it's lifetime. 
+`LoggerImpl` provides implementation of the singleton pattern.
+
+
+### Message loop thread in SDLLogger
+
+Message loop thread is needed to avoid significant performance degradation in run time as logging calls are blocking calls and might take too much time. 
```

Comment:
Message loop thread is needed to avoid significant performance degradation at run time as logging calls are blocking calls and might take a significant amount of time.

Reply:
Fixed with https://github.com/smartdevicelink/sdl_core_guides/pull/93/commits/23035bc00e4ad59156e472b85895ee08e898e47e